packages/scoutgame/src/builderNfts/refreshEstimatedPayouts.ts

```diff
@@ -0,0 +1,93 @@
+import { getPointsCountForWeekWithNormalisation } from '../points/getPointsCountForWeekWithNormalisation';
+
+import type { BuilderNftWithOwners, SeasonNfts } from './getAllSeasonNftsWithOwners';
+import { getAllSeasonNftsWithOwners } from './getAllSeasonNftsWithOwners';
+
+/**
+ * Refresh estimated payouts for a given week
+ *
+ * @param builderToRefresh - Builder ID to refresh. If not provided, all builders will be refreshed
+ */
+export async function refreshEstimatedPayouts({
+  week,
+  builderIdToRefresh
+}: {
+  week: ISOWeek;
+  builderIdToRefresh?: string;
+}): Promise<SeasonNfts> {
+  const season = getCurrentSeasonStart(week);
+
+  const { normalisedBuilders } = await getPointsCountForWeekWithNormalisation({
+    week
+  });
+
+  const data = await getAllSeasonNftsWithOwners({ season });
+
+  const seasonBuilderNfts = data.default.reduce(
+    (acc, nft) => ({
+      ...acc,
+      [nft.builderId]: nft
+    }),
+    {} as Record<string, BuilderNftWithOwners>
+  );
+
+  const seasonStarterPackNfts = data.starter_pack.reduce(
+    (acc, nft) => ({
+      ...acc,
+      [nft.builderId]: nft
+    }),
+    {} as Record<string, BuilderNftWithOwners>
+  );
+
+  for (const { builder, normalisedPoints } of normalisedBuilders) {
+    if (!builderIdToRefresh || builderIdToRefresh === builder.builder.id) {
+      const defaultNft = seasonBuilderNfts[builder.builder.id];
+      const starterPackNft = seasonStarterPackNfts[builder.builder.id];
+
+      const defaultNftBalance = defaultNft.nftOwners.reduce((acc, nft) => acc + nft.balance, 0);
+      const starterPackNftBalance = starterPackNft?.nftOwners.reduce((acc, nft) => acc + nft.balance, 0) ?? 0;
+
+      const builderNftBalanceWeighted = defaultNftBalance * nftTypeMultipliers.default;
+      const starterPackNftBalanceWeighted = starterPackNftBalance * nftTypeMultipliers.starter_pack;
+
+      const totalNftBalance = builderNftBalanceWeighted + starterPackNftBalanceWeighted;
+
+      const expectedPayoutForNextNftPurchase = Math.floor(
+        normalisedPoints / (totalNftBalance + nftTypeMultipliers.default)
+      );
+
+      const expectedPayoutForNextStarterPackPurchase = Math.floor(
+        normalisedPoints / (totalNftBalance + nftTypeMultipliers.starter_pack)
+      );
+
+      if (expectedPayoutForNextNftPurchase !== defaultNft.estimatedPayout) {
+        await prisma.builderNft.update({
+          where: {
+            id: defaultNft.id
+          },
+          data: {
+            estimatedPayout: expectedPayoutForNextNftPurchase
+          }
+        });
+      }
+
+      if (starterPackNft && expectedPayoutForNextStarterPackPurchase !== starterPackNft.estimatedPayout) {
+        await prisma.builderNft.update({
+          where: {
+            id: starterPackNft.id
+          },
+          data: {
+            estimatedPayout: expectedPayoutForNextStarterPackPurchase
+          }
+        });
+      }
+    }
+  }
+
+  return getAllSeasonNftsWithOwners({ season });
```

Comment:
this is probably a heavy method and there's no need for an update method to return data. the caller can just call this if they want

Reply:
Noted & fixed